### PR TITLE
Add storybook and a Header story

### PR DESCRIPTION
**Pull request by marcaaron**
### Details
Adds Storybook to E.cash

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/164021

### Tests
1. `npm i`
2. Run `npm run storybook`
3. Verify that storybook opens

### QA
1. Navigate to `https://staging.expensify.cash/docs`
1. Verify that storybook is accessible

### Tested On

- [X] Web
-  Mobile Web
-  Desktop
-  iOS
-  Android

### CODE REVIEW

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

**Comment by botify** [comment]
Hey, I noticed you changed some webpack configuration files. This can break production builds. Did you remember to run a production build locally to verify they still work?

File: babel.config.js [hunk 1]
```diff
@@ -32,6 +32,6 @@ const metro = {
 module.exports = ({caller}) => {
     // For `react-native` (iOS/Android) caller will be "metro"
     // For `webpack` (Web) caller will be "@babel-loader"
-    const runningIn = caller(({name}) => name);
+    const runningIn = caller((args = {}) => args.name);
```
Comment (roryabraham) [inline babel.config.js:35]: NAB but why was this necessary?

File: src/stories/Header.stories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Header from '../components/Header';
+
+export default {
```
Comment (roryabraham) [inline src/stories/Header.stories.js:3]: Can you explain what's going on in this file? Maybe a link to some docs? Should we add some instructions to the `STYLE.md` style guide for creating `*.stories.js` files?

**Review by roryabraham — COMMENTED** [review]
GH actions code seems fine!

File: babel.config.js (hunk 1 above)
Comment (marcaaron) [reply babel.config.js:35]: When storybook loads the babel config there is no `args` object so some things were quietly breaking. Specifically, parsing the docs out of `propTypes` would not work. Maybe I should add some comments so nobody breaks this.

File: src/stories/Header.stories.js (hunk 2 above)
Comment (marcaaron) [reply src/stories/Header.stories.js:3]: Up for that, but might choose to do it in a follow up. There's a lot of documentation about Storybook available so I'm not sure what would go the style guide at this point except maybe something like "head to the project site and learn more about how to create stories".

Comment (marcaaron) [reply src/stories/Header.stories.js:8]: Added some links to the docs. Could do something more substantial eventually.

File: babel.config.js (hunk 1 above)
Comment (marcaaron) [reply babel.config.js:36]: Added some comment here about the significance to SB

**Comment by shawnborton** [comment]
This is probably not the best place to deliver this feedback, but what is the difference between default and large for the header text size? I actually think large (assuming it's 17) is probably the default here. In fact, I think the only other time we would have a non-17 sized font in the header area is for chats. And I think that's okay - it's almost like we aren't using a font-size for the header title but rather, we are using a component as the header title - with the component being the little avatar(single, two, or counter) + name(s) component we use.

**Comment by marcaaron** [comment]
Oh hmm yeah that's a good point @shawnborton. Looking through the code now and I don't really see anything passing this prop at all. I think all of the headers are set to have "large" text.

**Comment by marcaaron** [comment]
Actually this is only true for `HeaderWithCloseButton`

https://github.com/Expensify/Expensify.cash/blob/66d265e144a69f0a40bf5ca05d1b506a5a19a4bd/src/components/HeaderWithCloseButton.js#L37-L45

**Comment by marcaaron** [comment]
e.g. the IOU view is not using the large text

Screen Shot 2021-05-14 at 7 33 40 AM
Screen Shot 2021-05-14 at 7 33 31 AM

**Comment by shawnborton** [comment]
Ah got it, the IOU view should be using the larger text size. I say we remove it as a prop and just update the default header styles everywhere to use 17px as the font size unless a component replaces the header title.

**Comment by marcaaron** [comment]
We could probably change this to `textSize="small"` and have the default be `17` if that makes more sense.

**Review by roryabraham — DISMISSED** [review]
Just tested locally and this is **AWESOME!!!** I'm stoked for this – is there a tracking issue to create stories for all our components? 😁

**Comment by marcaaron** [comment]
Fixed up that Header issue @shawnborton and added some stories for the more complex `HeaderWithCloseButton`. It shows off the variations:

https://user-images.githubusercontent.com/32969087/118315456-407b1d80-b491-11eb-8c85-20d3559f1357.mp4

**Comment by shawnborton** [comment]
Woo, thanks Marc!

File: .storybook/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+const dotenv = require('dotenv');
+const custom = require('../config/webpack/webpack.common');
+
+const env = dotenv.config({path: path.resolve(__dirname, '../.env.staging')}).parsed;
+
+module.exports = ({config}) => {
+    config.resolve.alias = {
+        'react-native-config': 'react-web-config',
+        'react-native$': 'react-native-web',
+        '@react-native-community/netinfo': './mock',
```
Comment (roryabraham) [inline .storybook/webpack.config.js:13]: Sorry if I missed this in a previous review, but can we just do this?

```suggestion
        '@react-native-community/netinfo': path.resolve(__dirname, '../__mocks__/@react-native-community/netinfo.js'),
```

Then you don't need to create `src/libs/NetInfo/mock.js`

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Comment by thienlnam** [comment]
Wowwww this is so cool

Screen Shot 2021-05-14 at 1 04 14 PM

File: .storybook/webpack.config.js (hunk 3 above)
Comment (marcaaron) [reply .storybook/webpack.config.js:13]: Won't really work because we are not running in `jest` and that mock depends on it. We could maybe not use the default mock tho with the jest.fn() lemme try and get back to ya.

Comment (marcaaron) [reply .storybook/webpack.config.js]: Alright this worked after getting rid of the jest dependency so I got rid of that other file :) Nice one!

**Review by thienlnam — APPROVED** [review]

**Review by roryabraham — APPROVED** [review]
Cool!

**Comment by isagoico** [comment]
@roryabraham Heyo! Are we able to test this? I tried navigating to https://staging.expensify.cash/docs and I was unable to access storybook.

**Comment by marcaaron** [comment]
@isagoico can check this off as we'll need to fix this somewhere else but can access via `/docs/index.html` for now. It's not a blocker in any case.
